diff --git a/exec.sh b/exec.sh
index 2af7156..86e9216 100644
--- a/exec.sh
+++ b/exec.sh
@@ -1,5 +1,5 @@
 #!/bin/sh
-pip install -r requirements.txt
-python setup.py build develop
+pip install -r /repo/requirements.txt
 
-CUDA_VISIBLE_DEVICES=0 python test.py --test_iter=1
\ No newline at end of file
+CUDA_VISIBLE_DEVICES=0 
+python /repo/experiments/geotransformer.kitti.stage5.gse.k3.max.oacl.stage2.sinkhorn/eval.sh 1
\ No newline at end of file
diff --git a/experiments/geotransformer.kitti.stage5.gse.k3.max.oacl.stage2.sinkhorn/eval.sh b/experiments/geotransformer.kitti.stage5.gse.k3.max.oacl.stage2.sinkhorn/eval.sh
index 8f3d797..324958f 100755
--- a/experiments/geotransformer.kitti.stage5.gse.k3.max.oacl.stage2.sinkhorn/eval.sh
+++ b/experiments/geotransformer.kitti.stage5.gse.k3.max.oacl.stage2.sinkhorn/eval.sh
@@ -1,3 +1,4 @@
+#!/bin/sh
 if [ "$2" = "test" ]; then
     python test.py --test_epoch=$1
 fi
