diff --git a/exec.sh b/exec.sh
index 2af7156..a8271cf 100644
--- a/exec.sh
+++ b/exec.sh
@@ -1,5 +1,12 @@
 #!/bin/sh
-pip install -r requirements.txt
-python setup.py build develop
+# [Optional] If you are using CUDA 11.0 or newer, please install `torch==1.7.1+cu110`
+pip install torch==1.7.1+cu110 -f https://download.pytorch.org/whl/torch_stable.html
 
-CUDA_VISIBLE_DEVICES=0 python test.py --test_iter=1
\ No newline at end of file
+# Install packages and other dependencies
+cd /repo
+python /repo/setup.py build develop -build-lib=/repo/
+pip install coloredlogs
+pip install einops
+
+CUDA_VISIBLE_DEVICES=0 
+python /repo/experiments/geotransformer.3dmatch.stage4.gse.k3.max.oacl.stage2.sinkhorn/demo.py --src_file=/mnt/vol21/i22_msbrf65/dataset/scan_data.ply --ref_file=/mnt/vol21/i22_msbrf65/dataset/map_data.ply --gt_file=/repo/dataset/T03.npy --weights=/repo/weights/geotransformer-kitti.pth.tar
\ No newline at end of file
diff --git a/experiments/geotransformer.3dmatch.stage4.gse.k3.max.oacl.stage2.sinkhorn/demo.py b/experiments/geotransformer.3dmatch.stage4.gse.k3.max.oacl.stage2.sinkhorn/demo.py
index 8664574..e0e7f0e 100644
--- a/experiments/geotransformer.3dmatch.stage4.gse.k3.max.oacl.stage2.sinkhorn/demo.py
+++ b/experiments/geotransformer.3dmatch.stage4.gse.k3.max.oacl.stage2.sinkhorn/demo.py
@@ -2,6 +2,7 @@ import argparse
 
 import torch
 import numpy as np
+import open3d
 
 from geotransformer.utils.data import registration_collate_fn_stack_mode
 from geotransformer.utils.torch import to_cuda, release_cuda
@@ -22,16 +23,16 @@ def make_parser():
 
 
 def load_data(args):
-    src_points = np.load(args.src_file)
-    ref_points = np.load(args.ref_file)
-    src_feats = np.ones_like(src_points[:, :1])
-    ref_feats = np.ones_like(ref_points[:, :1])
+    # src_points = np.load(args.src_file)
+    # ref_points = np.load(args.ref_file)
+    # src_feats = np.ones_like(src_points[:, :1])
+    # ref_feats = np.ones_like(ref_points[:, :1])
 
     data_dict = {
-        "ref_points": ref_points.astype(np.float32),
-        "src_points": src_points.astype(np.float32),
-        "ref_feats": ref_feats.astype(np.float32),
-        "src_feats": src_feats.astype(np.float32),
+        # "ref_points": ref_points.astype(np.float32),
+        # # "src_points": src_points.astype(np.float32),
+        # "ref_feats": ref_feats.astype(np.float32),
+        # "src_feats": src_feats.astype(np.float32),
     }
 
     if args.gt_file is not None:
@@ -66,26 +67,27 @@ def main():
     output_dict = release_cuda(output_dict)
 
     # get results
-    ref_points = output_dict["ref_points"]
-    src_points = output_dict["src_points"]
+    # ref_points = output_dict["ref_points"]
+    # src_points = output_dict["src_points"]
     estimated_transform = output_dict["estimated_transform"]
     transform = data_dict["transform"]
 
     # visualization
-    ref_pcd = make_open3d_point_cloud(ref_points)
+    ref_pcd = open3d.io.read_point_cloud("/mnt/vol21/i22_msbrf65/dataset/map_data.ply")
     ref_pcd.estimate_normals()
     ref_pcd.paint_uniform_color(get_color("custom_yellow"))
-    src_pcd = make_open3d_point_cloud(src_points)
+    src_pcd = open3d.io.read_point_cloud("/mnt/vol21/i22_msbrf65/dataset/scan_data.ply")
+    src_pcd.transform(transform)
     src_pcd.estimate_normals()
     src_pcd.paint_uniform_color(get_color("custom_blue"))
-    draw_geometries(ref_pcd, src_pcd)
+    # draw_geometries(ref_pcd, src_pcd)
     src_pcd = src_pcd.transform(estimated_transform)
-    draw_geometries(ref_pcd, src_pcd)
+    # draw_geometries(ref_pcd, src_pcd)
 
     # compute error
     rre, rte = compute_registration_error(transform, estimated_transform)
     print(f"RRE(deg): {rre:.3f}, RTE(m): {rte:.3f}")
-
+    np.save("/mnt/vol21/i22_msbrf65/dataset/T03_est_geo_kitti", estimated_transform)
 
 if __name__ == "__main__":
     main()
diff --git a/setup.py b/setup.py
index bfb00b9..134d4c7 100644
--- a/setup.py
+++ b/setup.py
@@ -9,12 +9,12 @@ setup(
         CUDAExtension(
             name='geotransformer.ext',
             sources=[
-                'geotransformer/extensions/extra/cloud/cloud.cpp',
-                'geotransformer/extensions/cpu/grid_subsampling/grid_subsampling.cpp',
-                'geotransformer/extensions/cpu/grid_subsampling/grid_subsampling_cpu.cpp',
-                'geotransformer/extensions/cpu/radius_neighbors/radius_neighbors.cpp',
-                'geotransformer/extensions/cpu/radius_neighbors/radius_neighbors_cpu.cpp',
-                'geotransformer/extensions/pybind.cpp',
+                '/repo/geotransformer/extensions/extra/cloud/cloud.cpp',
+                '/repo/geotransformer/extensions/cpu/grid_subsampling/grid_subsampling.cpp',
+                '/repo/geotransformer/extensions/cpu/grid_subsampling/grid_subsampling_cpu.cpp',
+                '/repo/geotransformer/extensions/cpu/radius_neighbors/radius_neighbors.cpp',
+                '/repo/geotransformer/extensions/cpu/radius_neighbors/radius_neighbors_cpu.cpp',
+                '/repo/geotransformer/extensions/pybind.cpp',
             ],
         ),
     ],
